### PR TITLE
Cmr 4382

### PR DESCRIPTION
**Pull request by laurenfrederick**
Change entry title to use wildcard search. 

Also added new anomaly tests for a ticket that came through from users and support for umm-json in anomaly tests.

### CODE REVIEW

File: search-app/src/cmr/search/data/keywords_to_elastic.clj [hunk 1]
```diff
@@ -91,6 +91,19 @@
   (let [regex (process-keyword keyword)]
     {:regexp {field regex}}))
 
+(defn- keywords->regex-filter
+  "Create a filter for keyword searches that checks for a loose match on "
+  [field keywords boost]
+  {:weight boost
+   ;; Should the 'and' below actually be an 'or'? Investigate this as part of CMR-1329
+   :filter {:or (concat
```
Comment (oubiwann) [inline search-app/src/cmr/search/data/keywords_to_elastic.clj]: This is a nice coding pattern when used in such a way. 👍

File: search-relevancy-test/src/search_relevancy_test/anomaly_fetcher.clj [hunk 2]
```diff
@@ -61,11 +73,11 @@
         concept-id->format-map (formats-for-collections concept-ids)]
     (doseq [[concept-id fmt] concept-id->format-map]
       (let [directory (get format-string->directory fmt)
-            filename (str concept-id ".xml")
+            filename (str concept-id (file-extension fmt))
```
Comment (oubiwann) [inline search-relevancy-test/src/search_relevancy_test/anomaly_fetcher.clj]: Loving the concise `let` blocks :-)

File: search-app/src/cmr/search/data/keywords_to_elastic.clj [hunk 3]
```diff
@@ -91,6 +91,19 @@
   (let [regex (process-keyword keyword)]
     {:regexp {field regex}}))
 
+(defn- keywords->regex-filter
+  "Create a filter for keyword searches that checks for a loose match on "
```
Comment (oubiwann) [inline search-app/src/cmr/search/data/keywords_to_elastic.clj]: Looks like you got interrupted while writing the docstring ... I'm guessing you'll compare/update the docstring here and the one for `keywords->name-filter`?

File: search-app/src/cmr/search/data/keywords_to_elastic.clj [hunk 4]
```diff
@@ -91,6 +91,19 @@
...
+   ;; Should the 'and' below actually be an 'or'? Investigate this as part of CMR-1329
+   :filter {:or (concat
+                 (when-let [keywords (:keywords keywords)]
+                  [{:and (map (partial keyword-regexp-filter field) keywords)}])
+                 (when-let [field-keywords (:field-keywords keywords)]
+                  [{:or (concat
+                         (map (partial keyword-regexp-filter field) field-keywords)
+                         (map (partial keyword-regexp-filter field) field-keywords))}]))}})
+
 (defn- keywords->name-filter
```
Comment (oubiwann) [inline search-app/src/cmr/search/data/keywords_to_elastic.clj]: Comparing this docstring to the one above ... not sure if these needs to be updated too? (now that how entry-title filtering has changed).

File: search-app/src/cmr/search/data/keywords_to_elastic.clj (hunk 3 above)
Comment (laurenfrederick) [inline search-app/src/cmr/search/data/keywords_to_elastic.clj]: Wow... thank you for catching that!

File: search-app/src/cmr/search/data/keywords_to_elastic.clj (hunk 4 above)
Comment (laurenfrederick) [inline search-app/src/cmr/search/data/keywords_to_elastic.clj]: This function has not been updated. I think it's confusing looking at it in github.

**Pull request merged**